sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -2652,5 +2652,114 @@ def my_pipeline(x: Input[Artifact] = Artifact(
+        self.assertEqual(
+            pipeline_spec.root.dag.tasks['print-op'].component_ref.name,
+            'comp-print-op')
+        self.assertEqual(
+            pipeline_spec.root.dag.tasks['print-op-2'].component_ref.name,
+            'comp-print-op')
+
+        # check task specific input/ output information is not lost
+        self.assertEqual(
+            pipeline_spec.root.dag.tasks['print-op-2'].inputs
+            .parameters['message'].runtime_value.constant.string_value,
+            'different')
+
+    def test_task_specific_executor_case(self):
+
+        @dsl.component
+        def print_op(message: str = 'defualt'):
+            print(message)
+
+        @dsl.pipeline()
+        def my_pipeline():
+            task1 = print_op()
+            task2 = print_op(
+                message='different').set_cpu_limit('4').set_memory_limit('14Gi')
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            package_path = os.path.join(tempdir, 'pipeline.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=my_pipeline, package_path=package_path)
+            pipeline_spec = pipeline_spec_from_file(package_path)
+
+        self.assertEqual(len(pipeline_spec.components), 2)
+
+        # check both tasks component spec and executors exist
+        self.assertIn('comp-print-op', pipeline_spec.components)
+        self.assertIn(
+            'exec-print-op', pipeline_spec.deployment_spec.fields['executors']
+            .struct_value.fields)
+        self.assertIn('comp-print-op-2', pipeline_spec.components)
+        self.assertIn(
+            'exec-print-op-2', pipeline_spec.deployment_spec.fields['executors']
+            .struct_value.fields)
+
+        # check both component specs contains the correct executor label
+        self.assertEqual(
+            pipeline_spec.components['comp-print-op'].executor_label,
+            'exec-print-op')
+        self.assertEqual(
+            pipeline_spec.components['comp-print-op-2'].executor_label,
+            'exec-print-op-2')
+
+        # check tasks point to corrrect component spec reference
+        self.assertEqual(
+            pipeline_spec.root.dag.tasks['print-op'].component_ref.name,
+            'comp-print-op')
+        self.assertEqual(
+            pipeline_spec.root.dag.tasks['print-op-2'].component_ref.name,
+            'comp-print-op-2')
+
+        # check information of resource spec is not lost
+        self.assertEqual(
+            4.0, pipeline_spec.deployment_spec.fields['executors'].struct_value
+            .fields['exec-print-op-2'].struct_value['container']
+            ['resources'].fields['cpuLimit'].number_value)
+        self.assertEqual(
+            15.032385536, pipeline_spec.deployment_spec.fields['executors']
+            .struct_value.fields['exec-print-op-2'].struct_value['container']
+            ['resources'].fields['memoryLimit'].number_value)
+
```

Comment:
These tests look good. Does it also make sense to have a test for the pipeline-in-pipeline case? And `ComponentSpec`-level configuration, such as display name?